### PR TITLE
fix: [PL-41802]: Checked headers before parsing response as JSON

### PR DESCRIPTION
**Pull request by mayankVermaHarness**
### Summary

JIRA: https://harness.atlassian.net/browse/PL-41802

- Checked headers before parsing response as JSON
- Remove global 403 handling - not needed
- Remove `page-not-public` route - not needed

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by mayankVermaHarness** [comment]
retrigger titlecheck

**Review by dhruvUpadhyay07 — DISMISSED** [review]

**Review by devinder-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by dhruvUpadhyay07 — APPROVED** [review]

File: src/framework/app/App.tsx [hunk 1]
```diff
@@ -120,64 +119,55 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
...
-            .json()
-            .then(res => {
+          const clonedResponse = response.clone()
+          if (isJsonResponse(clonedResponse)) {
+            clonedResponse.json().then(res => {
               showError(res.message || TOO_MANY_REQUESTS_MESSAGE)
             })
-          break
-        }
-        case 403: {
```
Comment (mayankVermaHarness) [inline src/framework/app/App.tsx:170]: 403 use-case added 1 week back is no longer needed. RBAC framework itself will handle this.

File: src/modules/10-common/RouteDestinations.tsx [hunk 2]
```diff
@@ -86,10 +85,6 @@ export default (
       <RedirectToHome />
     </Route>
 
-    <Route exact path={routes.toPageNotPublic({ ...accountPathProps })}>
```
Comment (mayankVermaHarness) [inline src/modules/10-common/RouteDestinations.tsx:89]: Route not needed anymore.

File: src/modules/10-common/RouteDefinitions.ts [hunk 3]
```diff
@@ -67,7 +67,6 @@ const CV_HOME = `/cv/home`
 const routes = {
   toMainDashboard: withAccountId(() => '/main-dashboard'),
   toHome: withAccountId(() => '/home'),
-  toPageNotPublic: withAccountId(() => '/page-not-public'),
```
Comment (mayankVermaHarness) [inline src/modules/10-common/RouteDefinitions.ts:70]: Route not needed anymore.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by jenil-harness — APPROVED** [review]

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Pull request merged**